src/Compilers/Core/Portable/SourceGeneration/GeneratorAttribute.cs

```diff
@@ -11,6 +11,43 @@ namespace Microsoft.CodeAnalysis
     [AttributeUsage(AttributeTargets.Class)]
     public sealed class GeneratorAttribute : Attribute
     {
-        // https://github.com/dotnet/roslyn/issues/: we don't know if we'll keep this, but for now it lets us re-use the analyzer discovery mechanism
+        /// <summary>
+        /// The source languages to which this generator applies.  See <see cref="LanguageNames"/>.
+        /// </summary>
+        public string[] Languages { get; }
+
+        /// <summary>
+        /// Attribute constructor used to specify the attached class is a source generator that provides CSharp sources.
```

Comment:
```suggestion
        /// Attribute constructor used to specify the attached class is a source generator that provides CSharp sources.
```

Crazy aside: is it required to be a `class` or can it say be a `struct`?

Reply:
It *has* to be a class. We explicitly set the allowed location of the attribute to be class (see line 11) so loading via AnalyzerReference it can't ever be a struct. 

As a second defense we check that the generator is also an `ISourceGenerator`, so there's no way you can pass a struct via the generator driver in code either.